public/app/plugins/panel/stat/StatPanel.tsx

```diff
@@ -83,6 +87,28 @@ export class StatPanel extends PureComponent<PanelProps<StatPanelOptions>> {
   getValues = (): FieldDisplay[] => {
     const { data, options, replaceVariables, fieldConfig, timeZone } = this.props;
 
+    let globalRange: NumericRange | undefined = undefined;
+
+    for (let frame of data.series) {
+      for (let field of frame.fields) {
+        let { config } = field;
+        // mostly copied from fieldOverrides, since they are skipped during streaming
+        // Set the Min/Max value automatically
+        if (field.type === FieldType.number) {
+          if (field.state?.range) {
+            continue;
+          }
+          if (!globalRange && (!isNumber(config.min) || !isNumber(config.max))) {
+            globalRange = findNumericFieldMinMax(data.series);
+          }
+          const min = config.min ?? globalRange!.min;
+          const max = config.max ?? globalRange!.max;
+          field.state = field.state ?? {};
+          field.state.range = { min, max, delta: max! - min! };
+        }
+      }
+    }
```

Comment:
Given that this logic is needed in almost any streaming graph panel, is there a way we can move this to a shared function.

It feels so hacky to have this concern (and amount of code) exist in each panel.

Reply:
yes, i'll pull this out into a util fn